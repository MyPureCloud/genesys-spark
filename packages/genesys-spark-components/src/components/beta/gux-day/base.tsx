import { Element } from '@stencil/core';

export class GuxElement {
    @Element()
    root: HTMLElement;
}