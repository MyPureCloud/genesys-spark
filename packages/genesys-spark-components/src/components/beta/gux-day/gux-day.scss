@use '~genesys-spark/dist/scss/mixins.scss';

:host {
  display: inline-block;
  inline-size: var(--gse-ui-calendarMenu-day-date-size);
  block-size: var(--gse-ui-calendarMenu-day-input-height);
  font-family: var(--gse-ui-calendarMenu-date-currentText-fontFamily);
  font-size: var(--gse-ui-calendarMenu-date-defaultText-fontSize);
  font-style: normal;
  font-weight: var(--gse-ui-calendarMenu-date-defaultText-fontWeight);
  line-height: 32px;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
  border: none;
  border-radius: var(--gse-ui-calendarMenu-month-borderRadius);
  outline: none;
}

:host(:hover) {
  background-color: var(--gse-ui-calendarMenu-date-hover-backgroundColor);
}

:host(:focus-within) {
  border-radius: var(--gse-ui-calendarMenu-month-focusBorderRadius);
  outline: var(
      --gse-semantic-focusRing-default-border-width,
      var(--gse-semantic-focusOutline-md-borderWidth)
    )
    solid
    var(
      --gse-semantic-focusRing-default-border-color,
      var(--gse-semantic-border-focus)
    );
}

:host([aria-current='true']) {
  color: var(--gse-ui-calendarMenu-date-selected-foregroundColor);
  background-color: var(--gse-ui-calendarMenu-date-selected-backgroundColor);
}

:host([aria-disabled='true']),
:host(.gux-muted) {
  opacity: var(--gse-ui-calendarMenu-disabled-opacity);
}

// The button should have no styles
button {
  @include mixins.gux-empty-button;
}
// The button should not have a focus outline, it is handled with :focus-within
// on :host
button:focus {
  outline: none;
}
