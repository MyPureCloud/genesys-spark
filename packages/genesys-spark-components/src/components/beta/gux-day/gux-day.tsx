import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Prop,
  State,
  Watch
} from '@stencil/core';
import { getDesiredLocale } from 'i18n';
import * as sparkIntl from '../../../genesys-spark-utils/intl';

@Component({
  styleUrl: 'gux-day.scss',
  tag: 'gux-day',
  shadow: { delegatesFocus: true }
})
export class GuxDay {
  @Element()
  root: HTMLElement;

  /* The date of the day in ISO format, e.g. 2024-10-22 */
  @Prop()
  day: string;

  /* JS Date object, derived from the `day` attribute */
  @State()
  date: Date;

  /* Watcher to sync the internal date with the day attribute on changes */
  @Watch('day')
  onDateAttrChange(): void {
    this.date = new Date(this.day);
  }

  @Event({ eventName: 'guxdayselected' })
  daySelected: EventEmitter<string>;

  /* Formatter for displaying the numeric day of the month */
  dayFormatter: Intl.DateTimeFormat;

  /* Formatter for screen readers that will read the full date */
  readerFormatter: Intl.DateTimeFormat;

  async connectedCallback(): Promise<void> {
    this.date = new Date(this.day);
    const locale = getDesiredLocale(this.root);
    this.dayFormatter = sparkIntl.dateTimeFormat(locale, { day: 'numeric' });
    this.readerFormatter = sparkIntl.dateTimeFormat(locale);
  }

  render(): JSX.Element {
    return (
      <button onClick={() => this.daySelected.emit(this.day)} type="button">
        <slot>{this.dayFormatter.format(this.date)}</slot>
      </button>
    );
  }
}
