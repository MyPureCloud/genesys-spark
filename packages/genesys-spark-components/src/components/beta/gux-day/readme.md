# gux-day



<!-- Auto Generated Below -->


## Properties

| Property | Attribute | Description | Type     | Default     |
| -------- | --------- | ----------- | -------- | ----------- |
| `day`    | `day`     |             | `string` | `undefined` |


## Dependencies

### Used by

 - [gux-calendar-beta](../gux-calendar-beta/components/single)

### Graph
```mermaid
graph TD;
  gux-calendar-beta --> gux-day
  style gux-day fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
