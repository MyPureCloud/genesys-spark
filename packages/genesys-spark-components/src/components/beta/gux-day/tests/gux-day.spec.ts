import { newSpecPage } from '@test/specTestUtils';
import { GuxDay } from '../gux-day';

const components = [GuxDay];
const language = 'en';
const html = `
<gux-day-beta day="2025-03-05"></gux-day-beta>
`;

describe('gux-day', () => {
  it('should build', async () => {
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxDay);
  });

  it('renders', async () => {
    const page = await newSpecPage({ components, html, language });

    expect(page.root).toMatchSnapshot();
  });
});
