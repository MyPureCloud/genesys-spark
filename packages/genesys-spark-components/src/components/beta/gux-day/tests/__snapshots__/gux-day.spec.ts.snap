// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-day renders 1`] = `
<gux-day-beta day="2025-03-05">
  <template shadowrootmode="open" shadowrootdelegatesfocus>
    <button type="button">
      <slot>
        <span aria-hidden="true">
          5
        </span>
        <span class="gux-sr-only">
          March 5, 2025
        </span>
      </slot>
    </button>
  </template>
</gux-day-beta>
`;
